client/src/main/java/io/pravega/client/netty/impl/ConnectionFactoryImpl.java

```diff
@@ -49,6 +53,9 @@
     private EventLoopGroup group;
     private boolean nio = false;
     private final AtomicBoolean closed = new AtomicBoolean(false);
+    private final ScheduledExecutorService executor = Executors.newScheduledThreadPool(
+            Runtime.getRuntime().availableProcessors(),
```

Comment:
I'm not sure this is a good idea. This essentially means that we will have as many threads as cores, and we could end up competing with the application for CPU resources. We should at least make this configurable.